Add markdown generation from spec.md.

nazrhom opened the pull request:
Also add decriptions to endpoints in spec.yaml


[rendered markdown file](https://github.com/mlabs-haskell/query-layer-impl/blob/623a0c0368f94e93c6cd926f293b7330024245dd/cip-spec.md)

nazrhom committed:
Add markdown generation from spec.md.
Also add decriptions to endpoints in spec.yaml

klntsky commented on cip-spec.md:
````diff
@@ -0,0 +1,406 @@
+<summary>Show example: </summary>
+
+```
+{"utxos":[{"input":{"transaction_id":"3333333333333333333333333333333333333333333333333333333333333333","index":858993459},"output":{"address":"stake177stake177","amount":{"coin":"0000000000"},"script_ref":{"tag":"plutus_script","value":{"language":"plutus_v1/plutus_v1","bytes":"333333333333"}}}}]}
````
Instead of example values (that are not valid anyway), show the schema or a link to the position in the schema file.

nazrhom replied:
@klntsky I pushed an update inlining the full schema in the generated markdown. This unfortunately makes the file so big that even GH will not generate a markdown preview from it (locally, my editor takes almost a full minute to generate the preview :D).
I think inlining the schema (at least fully-resolved schemas) is a no-go. I like the idea of linking to the schema file we discussed and will try to explore some options with that (ideally we can link to an openapi visualisation of the schema, but even the raw file might be ok).
Another alternative could be to inline the schema without fully resolving it (maybe making the $refs actual links to the type definition in the schema?). If we can't link to external resources I might explore this option instead.

As far as the example values go, why do you say they are not valid, can you point to something specific? I generally feel those could be useful for users to copy and then substitute the values they need in

klntsky replied:
> I think inlining the schema (at least fully-resolved schemas) is a no-go

OK. Not resolving is probably fine.

nazrhom committed:
Add schemas to generated MD

nazrhom committed:
Simplify code and rename some variables

nazrhom committed:
Add special case for address when checking schemas for equality.

Address is changed by removing one type in conway. We add a special treatment for it to
prevent creating an unecessary anyOf

nazrhom committed:
Add links to openapi endpoint in spec

nazrhom committed:
Remove transaction submit endpoint

nazrhom committed:
Fix markdown generation

nazrhom committed:
Add npm start alias

nazrhom committed:
Merge pull request #2 from mlabs-haskell/nazrhom/address-specialcase

Nazrhom/address specialcase
